### ISSUE TITLE
Records with same properties from different factories aren't equal

### ISSUE THREAD

**Issue by NickLydon** [immutable-js/immutable-js#1734]
Identical records created using different factories return on `.equals()`:
v3.8.2: **true**
v4.0.0-rc.12: **false**
This change may be intentional, but I couldn't determine that from the documentation.
Here's some sample code:
```
import {Record} from 'immutable';

const factoryA = Record({ id: '' });
const factoryB = Record({ id: '' });

console.log('factoryA equals factoryA', factoryA().equals(factoryA()));
console.log('factoryA equals factoryB', factoryA().equals(factoryB()));
```
[and the stackblitz for it](https://stackblitz.com/edit/typescript-a6a2fy)

**Comment by bennyrw** [comment]
I've also run into this - is this an intentional change? If we're creating multiple records of the same type is the expected design pattern to cache record factories for reuse?

**Comment by jdeniau** [comment]
Thank you for your bug report. The immutable-js oss fork will soon be merged and a fix for this issue has been included in #1833. Once this PR is merged, this issue will be resolved in the main branch. We will then do our best to to release the `4.0.0` version.

Commit reference: https://github.com/immutable-js/immutable-js/pull/1833/commits/a9a1612897627a6ad1217bfa0a76203c95e710b1
